rules.d/60-persistent-hidraw.rules

```diff
@@ -0,0 +1,15 @@
+# do not edit this file, it will be overwritten on update
+
+ACTION=="remove", GOTO="persistent_hidraw_end"
+SUBSYSTEM!="hidraw", GOTO="persistent_hidraw_end"
+
+SUBSYSTEMS=="usb", ENV{ID_BUS}=="", IMPORT{builtin}="usb_id"
+
+# by-id links
+KERNEL=="hidraw*", ENV{ID_BUS}=="?*", ATTRS{bInterfaceNumber}=="|00", SYMLINK+="input/by-id/$env{ID_BUS}-$env{ID_SERIAL}-hidraw"
+KERNEL=="hidraw*", ENV{ID_BUS}=="?*", ATTRS{bInterfaceNumber}=="?*", ATTRS{bInterfaceNumber}!="00", SYMLINK+="input/by-id/$env{ID_BUS}-$env{ID_SERIAL}-if$attr{bInterfaceNumber}-hidraw"
+
+# add a more readable 'fido' link for devices with ID_FIDO_TOKEN==1
+KERNEL=="hidraw*", ENV{ID_BUS}=="?*", ENV{ID_FIDO_TOKEN}=="?*", SYMLINK+="input/by-id/$env{ID_BUS}-$env{ID_SERIAL}-fido-hidraw"
```

Comment:
hmm, i am a bit concerned not to do the bInterfaceNumber here... i think usb composite devices can have multiple interfaces of the same device type, and thus there might be a single device that exposes multiple fido2 devices, at least in theory...

hence, i think we should do the same stuff as above, i.e. suppress the interface part on interface 00 or if none is defined, but otherwise include it in the symlink name

Reply:
Okay, makes sense. I took a different approach that still adds "fido" to the symlink.